client/src/components/Map/UCIMap.js

```diff
@@ -72,51 +239,56 @@ export default class UCIMap extends PureComponent {
+                    )
             )
-                return;
+            .sort((event, event2) => event.start - event2.start)
+            .forEach((event) => {
+                // Get building code, get id of building code, which will get us the building data from buildingCatalogue
+                const buildingCode = event.bldg.split(' ').slice(0, -1).join(' ');
+                const id = locations[buildingCode];
+                const locationData = buildingCatalogue[id];
+                const courseString = `${event.title} ${event.sectionType} @ ${event.bldg}`;
 
-            // Get building code, get id of building code, which will get us the building data from buildingCatalogue
-            const buildingCode = event.bldg.split(' ')[0];
-            const id = locations[buildingCode];
-            const locationData = buildingCatalogue[id];
-            const courseString = `${event.title} ${event.sectionType} @ ${event.bldg}`;
+                index++; // index temp fix for courses in same building
+                if (locationData === undefined || pinnedCourses.has(courseString)) return;
 
-            if (locationData === undefined || pinnedCourses.has(courseString)) return;
+                // Acronym, if it exists, is in between parentheses
+                const acronym = locationData.name.substring(
+                    locationData.name.indexOf('(') + 1,
+                    locationData.name.indexOf(')')
+                );
 
-            // Acronym, if it exists, is in between parentheses
-            const acronym = locationData.name.substring(
-                locationData.name.indexOf('(') + 1,
-                locationData.name.indexOf(')')
-            );
-
-            pinnedCourses.add(courseString);
-
-            markers.push(
-                <MapMarkerPopup
-                    image={locationData.imageURLs[0]}
-                    markerColor={event.color}
-                    location={locationData.name}
-                    lat={locationData.lat}
-                    lng={locationData.lng}
-                    acronym={acronym}
-                >
-                    <Fragment>
-                        <hr />
-                        Class: {`${event.title} ${event.sectionType}`}
-                        <br />
-                        Room: {event.bldg.split(' ')[1]}
-                    </Fragment>
-                </MapMarkerPopup>
-            );
-        });
+                pinnedCourses.add(courseString);
 
+                markers.push(
+                    <MapMarkerPopup
+                        image={locationData.imageURLs[0]}
+                        markerColor={event.color}
+                        location={locationData.name}
+                        lat={locationData.lat}
+                        lng={locationData.lng}
+                        acronym={acronym}
+                        index={this.state.day ? (index + 1).toString() : ''}
```

Comment:
Since you increment before generating the MapMarkerPopup, we can remove the +1. This should fix the bug where all the classes start at 2
```suggestion
                        index={this.state.day ? index.toString() : ''}
```